java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonMerge

Facing error while running test case with Junit 5 when using the schema registry mock:

    [INFO] -------------------------------------------------------
    [INFO]  T E S T S
    [INFO] -------------------------------------------------------
    [INFO] Running com.loconav.IgnitionAlertTest
    [main] INFO org.eclipse.jetty.util.log - Logging initialized @1028ms
    [main] INFO org.eclipse.jetty.server.Server - jetty-9.2.24.v20180105
    [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@4566d049{/__admin,null,AVAILABLE}
    [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@7f6f61c8{/,null,AVAILABLE}
    [main] INFO org.eclipse.jetty.server.NetworkTrafficServerConnector - Started NetworkTrafficServerConnector@4a11eb84{HTTP/1.1}{0.0.0.0:51339}
    [main] INFO org.eclipse.jetty.server.Server - Started @1225ms
    [main] INFO io.confluent.kafka.serializers.KafkaAvroSerializerConfig - KafkaAvroSerializerConfig values: 
            schema.registry.url = [http://localhost:51339]
            basic.auth.user.info = [hidden]
            auto.register.schemas = true
            max.schemas.per.subject = 1000
            basic.auth.credentials.source = URL
            schema.registry.basic.auth.user.info = [hidden]
            value.subject.name.strategy = class io.confluent.kafka.serializers.subject.TopicNameStrategy
            key.subject.name.strategy = class io.confluent.kafka.serializers.subject.TopicNameStrategy

    [main] INFO org.eclipse.jetty.server.NetworkTrafficServerConnector - Stopped NetworkTrafficServerConnector@4a11eb84{HTTP/1.1}{0.0.0.0:0}
    [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Stopped o.e.j.s.ServletContextHandler@7f6f61c8{/,null,UNAVAILABLE}
    [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Stopped o.e.j.s.ServletContextHandler@4566d049{/__admin,null,UNAVAILABLE}
    [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.823 s <<< FAILURE! - in com.loconav.IgnitionAlertTest
    [ERROR] alertIsGeneratedWithCorrectTimestampForDeviceCoordinates  Time elapsed: 0.237 s  <<< ERROR!
    java.lang.NoClassDefFoundError: com/fasterxml/jackson/annotation/JsonMerge
            at com.loconav.IgnitionAlertTest.setup(IgnitionAlertTest.java:52)
    Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonMerge
            at com.loconav.IgnitionAlertTest.setup(IgnitionAlertTest.java:52)

    [INFO] 
    [INFO] Results:
    [INFO] 
    [ERROR] Errors: 
    [ERROR]   IgnitionAlertTest.setup:52 ? NoClassDefFound com/fasterxml/jackson/annotation/...
    [INFO] 
    [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

The above mentioned line number is when I am retrieving `Topology` from the main code. Line no 51,52,53 in Test:

    50    IgnitionAlertService alertService = new IgnitionAlertService();
    51    Topology topology =
    52        alertService.generateTopology(
    53            props.getProperty(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG));

I am not using `JsonMerge` annotation anywhere in my main code or test cases.  Here is the dependency tree:

    [INFO] Scanning for projects...
    [INFO] 
    [INFO] --------------------< com.loconav:ignition-alerts >---------------------
    [INFO] Building ignition-alerts 1.0.0
    [INFO] --------------------------------[ jar ]---------------------------------
    [INFO] 
    [INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ ignition-alerts ---
    [INFO] com.loconav:ignition-alerts:jar:1.0.0
    [INFO] +- com.loconav:kafka-clients-core:jar:1.0.0:compile
    [INFO] |  +- org.apache.kafka:kafka-clients:jar:2.1.0:compile
    [INFO] |  |  +- com.github.luben:zstd-jni:jar:1.3.5-4:compile
    [INFO] |  |  +- org.lz4:lz4-java:jar:1.5.0:compile
    [INFO] |  |  +- org.xerial.snappy:snappy-java:jar:1.1.7.2:compile
    [INFO] |  |  \- org.slf4j:slf4j-api:jar:1.7.25:compile
    [INFO] |  +- org.apache.kafka:kafka-streams:jar:2.1.0:compile
    [INFO] |  |  +- org.apache.kafka:connect-json:jar:2.1.0:compile
    [INFO] |  |  |  \- org.apache.kafka:connect-api:jar:2.1.0:compile
    [INFO] |  |  |     \- javax.ws.rs:javax.ws.rs-api:jar:2.1.1:compile
    [INFO] |  |  \- org.rocksdb:rocksdbjni:jar:5.14.2:compile
    [INFO] |  +- io.confluent:kafka-schema-registry-client:jar:5.1.2:compile
    [INFO] |  |  +- io.confluent:common-config:jar:5.1.2:compile
    [INFO] |  |  +- org.apache.avro:avro:jar:1.8.1:compile
    [INFO] |  |  |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:compile
    [INFO] |  |  |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
    [INFO] |  |  |  +- com.thoughtworks.paranamer:paranamer:jar:2.7:compile
    [INFO] |  |  |  +- org.apache.commons:commons-compress:jar:1.8.1:compile
    [INFO] |  |  |  \- org.tukaani:xz:jar:1.5:compile
    [INFO] |  |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.9.6:compile
    [INFO] |  |  \- io.confluent:common-utils:jar:5.1.2:compile
    [INFO] |  |     +- org.apache.zookeeper:zookeeper:jar:3.4.13:compile
    [INFO] |  |     |  +- jline:jline:jar:0.9.94:compile
    [INFO] |  |     |  +- org.apache.yetus:audience-annotations:jar:0.5.0:compile
    [INFO] |  |     |  \- io.netty:netty:jar:3.10.6.Final:compile
    [INFO] |  |     \- com.101tec:zkclient:jar:0.10:compile
    [INFO] |  +- com.loconav:kafka-schemas:jar:1.0.0:compile
    [INFO] |  |  \- com.google.guava:guava:jar:27.1-jre:compile
    [INFO] |  |     +- com.google.guava:failureaccess:jar:1.0.1:compile
    [INFO] |  |     +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
    [INFO] |  |     +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
    [INFO] |  |     +- org.checkerframework:checker-qual:jar:2.5.2:compile
    [INFO] |  |     +- com.google.errorprone:error_prone_annotations:jar:2.2.0:compile
    [INFO] |  |     +- com.google.j2objc:j2objc-annotations:jar:1.1:compile
    [INFO] |  |     \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.17:compile
    [INFO] |  +- com.loconav:kafka-streams-services:jar:1.0.0:compile
    [INFO] |  \- org.slf4j:slf4j-simple:jar:1.7.25:compile
    [INFO] +- org.junit.jupiter:junit-jupiter-engine:jar:5.4.2:test
    [INFO] |  +- org.apiguardian:apiguardian-api:jar:1.0.0:test
    [INFO] |  \- org.junit.platform:junit-platform-engine:jar:1.4.2:test
    [INFO] +- org.junit.jupiter:junit-jupiter-api:jar:5.4.2:test
    [INFO] |  +- org.opentest4j:opentest4j:jar:1.1.1:test
    [INFO] |  \- org.junit.platform:junit-platform-commons:jar:1.4.2:test
    [INFO] +- com.bakdata.fluent-kafka-streams-tests:schema-registry-mock:jar:1.0.1:compile
    [INFO] |  +- io.confluent:kafka-avro-serializer:jar:5.1.0:compile
    [INFO] |  +- io.confluent:kafka-streams-avro-serde:jar:5.1.0:compile
    [INFO] |  \- com.github.tomakehurst:wiremock:jar:2.20.0:runtime
    [INFO] |     +- org.eclipse.jetty:jetty-server:jar:9.2.24.v20180105:runtime
    [INFO] |     |  +- javax.servlet:javax.servlet-api:jar:3.1.0:runtime
    [INFO] |     |  +- org.eclipse.jetty:jetty-http:jar:9.2.24.v20180105:runtime
    [INFO] |     |  \- org.eclipse.jetty:jetty-io:jar:9.2.24.v20180105:runtime
    [INFO] |     +- org.eclipse.jetty:jetty-servlet:jar:9.2.24.v20180105:runtime
    [INFO] |     |  \- org.eclipse.jetty:jetty-security:jar:9.2.24.v20180105:runtime
    [INFO] |     +- org.eclipse.jetty:jetty-servlets:jar:9.2.24.v20180105:runtime
    [INFO] |     |  +- org.eclipse.jetty:jetty-continuation:jar:9.2.24.v20180105:runtime
    [INFO] |     |  \- org.eclipse.jetty:jetty-util:jar:9.2.24.v20180105:runtime
    [INFO] |     +- org.eclipse.jetty:jetty-webapp:jar:9.2.24.v20180105:runtime
    [INFO] |     |  \- org.eclipse.jetty:jetty-xml:jar:9.2.24.v20180105:runtime
    [INFO] |     +- com.fasterxml.jackson.core:jackson-core:jar:2.8.11:compile
    [INFO] |     +- com.fasterxml.jackson.core:jackson-annotations:jar:2.8.11:compile
    [INFO] |     +- org.apache.httpcomponents:httpclient:jar:4.5.5:runtime
    [INFO] |     |  +- org.apache.httpcomponents:httpcore:jar:4.4.9:runtime
    [INFO] |     |  +- commons-logging:commons-logging:jar:1.2:runtime
    [INFO] |     |  \- commons-codec:commons-codec:jar:1.10:runtime
    [INFO] |     +- org.xmlunit:xmlunit-core:jar:2.5.1:runtime
    [INFO] |     +- org.xmlunit:xmlunit-legacy:jar:2.5.1:runtime
    [INFO] |     +- com.jayway.jsonpath:json-path:jar:2.4.0:runtime
    [INFO] |     |  \- net.minidev:json-smart:jar:2.3:runtime
    [INFO] |     |     \- net.minidev:accessors-smart:jar:1.2:runtime
    [INFO] |     |        \- org.ow2.asm:asm:jar:5.0.4:runtime
    [INFO] |     +- net.sf.jopt-simple:jopt-simple:jar:5.0.3:runtime
    [INFO] |     +- junit:junit:jar:4.12:runtime
    [INFO] |     +- org.apache.commons:commons-lang3:jar:3.7:runtime
    [INFO] |     +- com.flipkart.zjsonpatch:zjsonpatch:jar:0.4.4:runtime
    [INFO] |     +- com.github.jknack:handlebars:jar:4.0.7:runtime
    [INFO] |     |  \- org.antlr:antlr4-runtime:jar:4.7.1:runtime
    [INFO] |     \- com.github.jknack:handlebars-helpers:jar:4.0.7:runtime
    [INFO] +- org.apache.kafka:kafka-streams-test-utils:jar:2.1.0:compile
    [INFO] +- org.hamcrest:hamcrest-all:jar:1.3:test
    [INFO] \- joda-time:joda-time:jar:2.10.1:compile
    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD SUCCESS
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time:  1.876 s
    [INFO] Finished at: 2019-04-09T14:53:00+05:30
    [INFO] ------------------------------------------------------------------------

[Here](https://pastebin.com/0R6eug6s) is the full debug log while running `mvn -X test`. [This](https://pastebin.com/MnsrMYAZ) link contains the effective pom.

--- a/schema-registry-mock/build.gradle.kts
+++ b/schema-registry-mock/build.gradle.kts
@@ -12,4 +12,10 @@ dependencies {
     "api"(group = "io.confluent", name = "kafka-streams-avro-serde", version = confluentVersion)
 
     implementation(group = "com.github.tomakehurst", name = "wiremock", version = "2.20.0")
+
+    // resolve dependency issues between wiremock and schema registry
+    val jacksonVersion = "2.9.6"
+    implementation(group = "com.fasterxml.jackson.core", name = "jackson-core", version = jacksonVersion)
+    implementation(group = "com.fasterxml.jackson.core", name = "jackson-annotations", version = jacksonVersion)
+    implementation(group = "com.fasterxml.jackson.core", name = "jackson-databind", version = jacksonVersion)
 }

